staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -136,22 +135,18 @@ func (f *structuredMergeManager) Update(liveObj, newObj runtime.Object, managed
 }
 
 // Apply implements Manager.
-func (f *structuredMergeManager) Apply(liveObj runtime.Object, patch []byte, managed Managed, manager string, force bool) (runtime.Object, Managed, error) {
-	patchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
-	if err := yaml.Unmarshal(patch, &patchObj.Object); err != nil {
-		return nil, nil, errors.NewBadRequest(fmt.Sprintf("error decoding YAML: %v", err))
-	}
-
+func (f *structuredMergeManager) Apply(liveObj, patchObj runtime.Object, managed Managed, manager string, force bool) (runtime.Object, Managed, error) {
 	// Check that the patch object has the same version as the live object
-	if patchObj.GetAPIVersion() != f.groupVersion.String() {
+	if patchObj.GetObjectKind().GroupVersionKind().GroupVersion() != f.groupVersion {
 		return nil, nil,
 			errors.NewBadRequest(
 				fmt.Sprintf("Incorrect version specified in apply patch. "+
 					"Specified patch version: %s, expected: %s",
-					patchObj.GetAPIVersion(), f.groupVersion.String()))
+					patchObj.GetObjectKind().GroupVersionKind().GroupVersion(), f.groupVersion))
 	}
 
-	if patchObj.GetManagedFields() != nil {
+	patchObjMeta, err := meta.Accessor(patchObj)
```

Comment:
You're not doing anything with `err`, can you either panic, return or ignore altogether (`_`)?